scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1658,10 +1750,12 @@ private void sendMessage(final RawData message) throws HandshakeException {
 					sessionId = connection.getSessionIdentity();
 				}
 				// terminate the previous connection and add the new one to the store
-				Connection newConnection = new Connection(peerAddress);
+				Connection newConnection = new Connection(peerAddress, connection.getExecutor());
 				connection.cancelPendingFlight();
+				synchronized (connectionStore) {
```

Comment:
should we add a new atomic function in connectionStore ? (e.g. `replace` ?)

Reply:
May be at the end of the "connection id wip".